backend/prismaCalls.ts

```diff
@@ -42,4 +43,65 @@ export async function getPackageHistories(queryName: apiSchema.PackageName){
         console.log(`Error in getPackageHistories: ${error}`);
         return null;
     }
+}
+// For download endpoint
+export async function getPackageWithMetadataAndData(queryName: string, version: string): Promise<{ metadata: prismaSchema.PackageMetadata, data: prismaSchema.PackageData } | null> {
+    try {
+        const packageEntry = await prisma.package.findFirst({
+            where: {
+                metadata: {
+                    name: queryName,
+                    version: version
+                }
+            },
+            include: {
+                metadata: true,
+                data: true
+            }
+        });
+
+        if (!packageEntry || !packageEntry.data || !packageEntry.metadata) {
+            return null;
+        }
+
+        return {
+            metadata: packageEntry.metadata,
+            data: packageEntry.data
+        };
```

Comment:
returning explicit metaData and data doesn't make sense unless your packageEntry variable doesn't conform to the schema? But that's why you have the map in the backend function. So just return the packageEntry and use a map in the backend function.